src/main/kotlin/com/squareup/cash/hermit/ProjectExtensions.kt

```diff
@@ -130,4 +131,25 @@ private fun Project.runHermit(vararg args: String): Result<Process> {
     } else {
         success(process)
     }
+}
+
+
+/**
+ * HACK:
+ * This is using reflection, as the API is not available publicly yet.
+ * https://github.com/cashapp/hermit-ij-plugin/issues/37
+ */
+fun Project.isTrustedForHermit(): ThreeState {
+    try {
+        val trustedProjects = Class.forName("com.intellij.ide.impl.TrustedProjects")
+        val isTrusted = trustedProjects.methods.find { it.name == "isTrusted" }!!
+        val result = isTrusted.invoke(trustedProjects, this)
+        if (result !is Boolean) {
+            return ThreeState.UNSURE
+        }
+
+        return ThreeState.fromBoolean(result)
+    } catch (e: Exception) {
```

Comment:
Maybe we should specify the expected exceptions here instead a blanket catch all? This could potentially hide real bugs.

Reply:
Thinking more about this, should/not we want to hide these bugs?

* In the current scenario, any bug on our side will just cause the project to remain disabled, and prompt the user for explicit action.
* If we expose these bugs, then this will propagate to the user during project opening, which is not really actionable/informative to him.

Should we still hide it, and log them so that we can review after roll out if we need to make any changes? are these logs uploaded somewhere we can query?

Reply:
I think it is better to have logic bugs to crash as soon as possible so they get fixed. This would give the users a stack trace in their logs, which would be good information to us. I don't think we should try to replace IntelliJ error reporting with our own

Reply:
The logs are not automatically uploaded, unfortunately. We will have to ask users to send them over if they are having issues